### ISSUE TITLE
SVG: Could not import SVG that contains multibyte sequence

### ISSUE THREAD

**Issue by totake0224** [FreeCAD/FreeCAD#18458]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

Hello.
FreeCad could not import SVG file that contain multibyte sequence in property named "aria-label".
it output error like this 
<class 'UnicodeDecodeError'>: 'cp932' codec can't decode byte 0x88 in position 56970: illegal multibyte sequence
(whole message is attached)
[freecad error.txt](https://github.com/user-attachments/files/18122035/freecad.error.txt)

to reproduce this. I prepare small SVG file named "multibyte_in_aria-label.svg"
![multibyte_in_aria-label](https://github.com/user-attachments/assets/f64e94d0-6b4d-46f3-98bd-65fc89453700)

and I also prepare SVG file remove multibyte in aria-label named "remove_multibyte_in_aria-label.svg"
just remove multibyte from aria-label property. and this file, freecad import finish successfully.
![remove_multibyte_in_aria-label](https://github.com/user-attachments/assets/8c15cc49-23d7-43d1-a8c5-6df82a14e343)

### Full version info

```shell
OS: Windows 11 build 26100
Architecture: x86_64
Version: 1.0.0.39109 (Git) Conda
Build type: Release
Branch: (HEAD detached at 1.0.0)
Hash: 2fcc5317fe3aee96ca73475986a577719fc78e20
Python 3.11.10, Qt 5.15.15, Coin 4.0.3, Vtk 9.3.0, OCC 7.8.1
Locale: Japanese/Japan (ja_JP)
Stylesheet/Theme/QtStyle: FreeCAD Light.qss/FreeCAD Light/Fusion
```

### Subproject(s) affected?

None

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by totake0224** [comment]
In addition to this.
FreeCad also output same error multibye in propety "sodipodi:docname"

**Comment by maxwxyz** [comment]
How do you import it?

**Comment by totake0224** [comment]
at Draft mode.
use File->import menu.
I think mode is not matter.

**Comment by totake0224** [comment]
please save images at first post and check import function.
2 images look same but first one is contain multi-bye string and second doesn't.

**Comment by totake0224** [comment]
I found what is the problem.
This problem may occur only Japanese Windows environment.
explanation.
InkScape output utf-8 SVG file on windows, but
Freecad use pyopen without encoding parameter in Mod/Draft/ImportSVG.py.
like
> line 1788 : parser.parse(pyopen(filename)) 

So pyopen use default encoding CP932 that is python default encoding in Japanese Windows.
then it's fail.
I fixed this by adding encoding to 'pyopen' like,
> line 1788 : parser.parse(pyopen(filename,encoding='utf-8'))
I think it's more better that detecting file charset before pyopen. but I could not find way to install python library 'chardet' which provide charset detection.

**Comment by totake0224** [comment]
I think encoding='utf-8' is OK.
because I believe utf-8 is default encoding of MAC and Linux and Windows(10 and 11) GUI application.
(Somehow Japanese Windows still use CP932 on console application)
but there is another pyopen in FreeCad. So I suggest use UTF8 Mode of python . it enabled by setting environment value PYTHONUTF8=1.
I will confirm this option work or not.

I'm very happy if my work will be useful for FreeCad community.

**Comment by totake0224** [comment]
PYTHONUTF8 option doesn't work.
I set PYTHONUTF8 at Windows control panel.
but I can not confirm the value on FreeCad Python console.
I'm not Python user. it difficult to me finding the problem.
so I think encoding='utf-8' is only solution so far.
Is it need posting pull request?

**Comment by luzpaz** [comment]
Yes, please make a Pull Request so you can get the recognition for finding and fixing the bug :+1:

**Comment by luzpaz** [comment]
@chennes should we use the same strategy for other [instances of pyopen](https://github.com/search?q=repo%3AFreeCAD%2FFreeCAD%20pyopen&type=code) (as @totake0224 has pointed out) ?

**Comment by luzpaz** [comment]
Can we make a CI check or lint check for this?

@totake0224 any interest in making a 2nd PR that implements this change across the codebase?

**Comment by totake0224** [comment]
I'm interested in contribute something to open-source community.
But I'm not well trained software engineer. so it will take a time to complete.
If it's okay taking a time to complete, I'd like to do it.

Frankly speaking, Yesterday, it's my first time download FreeCad source.
I now preparing build environment of FreeCad.

**Comment by totake0224** [comment]
Hi,
I'm considering how to add encoding＝'utf-8' to pyopen.
and I would like to have advice from another developer.
how about this code.
original source line 64.(importSVG.py)
```
from builtins import open as pyopen
```
I changed this to
```
import builtins
#redefine pyopen as open with encoding='utf-8'
def utf8_open(file, mode='r', buffering=-1, encoding=None, errors=None, newline=None, closefd=True, opener=None):
    if encoding is None:
        encoding = 'utf-8'
    return builtins.open(file, mode, buffering, encoding, errors, newline, closefd, opener)
pyopen = utf8_open
```
I confirmed this works fine.

**Comment by luzpaz** [comment]
@chennes 👆

**Comment by luzpaz** [comment]
bumping again so this doesn't fall to the wayside

**Comment by totake0224** [comment]
I could not get the meaning [bumping again so this doesn't fall to the wayside]
For now,　I would create pull request.

**Comment by luzpaz** [comment]
@totake0224 'bump' in the context of 'bumping a ticket' is like 'poking' someone to get their attention ;)

Thanks you for the PR!

**Comment by luzpaz** [comment]
@chennes should I open a separate ticket to cover the other pyopen instances per https://github.com/FreeCAD/FreeCAD/issues/18458#issuecomment-2591571659 ?

**Comment by totake0224** [comment]
I will make another PR associated to this issue. after first PR would be accepted. 
Because I think it difficult to say which function will be affected by pyopen.
